pkg/gofr/config/godotenv_test.go

```diff
@@ -99,20 +99,23 @@ func Test_EnvFailureWithHyphen(t *testing.T) {
 
 	logger := logging.NewMockLogger(logging.DEBUG)
 
-	err := createConfigsDirectory()
-	if err != nil {
-		t.Error(err)
-	}
+	configFiles := []string{".env", ".local.env"}
 
-	// Call the function to create the .env file
-	createEnvFile(t, ".env", envData)
+	for _, file := range configFiles {
+		err := createConfigsDirectory()
+		if err != nil {
+			t.Error(err)
+		}
 
-	defer os.RemoveAll("configs")
+		createEnvFile(t, file, envData)
```

Comment:
You removed the defer cleanup.

You could add a t.Cleanup of the file into createEnvFile

Or maybe the configs folder is already cleaned?

I would add the t.Cleanup to createEnvFile as it could be reused

Reply:
Another solution is to refactor the code of createEnvFile to path a folder as argument instead of hardcoded "configs" twice.

then to use t.TempDir to generate a folder, then pass it to createEnvFile and

Reply:
This would be the best solution according to me

Reply:
Please note this could be addressed in another PR by creating a new backlog issue

Reply:
I created

- https://github.com/gofr-dev/gofr/issues/857